crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -320,14 +460,14 @@ async fn handle_evm_set_next_block_timestamp(
                         error_response_data(0, &format!("Timestamp {timestamp:?} is equal to the previous block's timestamp. Enable the 'allowBlocksWithSameTimestamp' option to allow this"))
                     } else {
                         let mut next_block_timestamp = state.next_block_timestamp.write().await;
-                        *next_block_timestamp = timestamp.into();
+                        *next_block_timestamp = Some(timestamp.into());
                         ResponseData::Success {
-                            result: next_block_timestamp.to_string(),
+                            result: next_block_timestamp.unwrap().to_string(),
```

Comment:
Why not use `timestamp` or an intermediate result thereof? The `.unwrap` seems unnecessary here

Reply:
Thanks, addressed in https://github.com/NomicFoundation/hardhat/pull/4243/commits/015bf7d7b1bc43dcbb6b6e3fe5c0524bc972ae00